cpp/src/io/json/reader_impl.cu

```diff
@@ -469,41 +469,50 @@ void reader::impl::set_data_types(cudaStream_t stream)
   if (!dtype.empty()) {
     CUDF_EXPECTS(dtype.size() == metadata.column_names.size(),
                  "Need to specify the type of each column.\n");
+
     // Assume that the dtype is in dictionary format only if all elements contain a colon
-    const bool is_dict = std::all_of(dtype.begin(), dtype.end(), [](const std::string &s) {
-      return std::find(s.begin(), s.end(), ':') != s.end();
-    });
+    const bool is_dict =
+      std::all_of(std::cbegin(dtype), std::cend(dtype), [](const std::string &s) {
+        return std::find(s.begin(), s.end(), ':') != s.end();
```

Comment:
Why std::cbegin (free func) above but s.begin (member) here?

Reply:
My rule is always preferring the free functions, because effectively the free function will call the member function when it exists, but there are some cases where the member functions don't exist (so you obviously can't use them) but the free functions still work. So preferring the free functions is my guideline that I follow because it is more general and I like being consistent. 

----
After googling for where the guideline came from (I wasn't the one that came up with this I just heard about it at some point), I found that it was actually Herb Sutter (chair of the ISO Committee) popularized it: 
https://stackoverflow.com/questions/7593086/why-use-non-member-begin-and-end-functions-in-c11/7593152

Reply:
I am aware of Sutter's guideline. But why not use std::cbegin(s) in the line I commented on?

Reply:
:facepalm: sighs, i should really learn to read better - changes pushed.